pkg/controller/provider/container/ovirt/client.go

```diff
@@ -226,3 +234,365 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
+	}
+
+	// Get a disk identified by uuid
+	diskService := disksService.DiskService(diskIdOVF)
+	diskRequest := diskService.Get()
+	diskRequestSent, err := diskRequest.Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	disk, ok := diskRequestSent.Disk()
+
+	diskId, ok := disk.Id()
+
+	//in case the OVF disk is locked wait fo OK status befor stating the image transfer
+	conn.WaitForDisk(diskId, ovirtsdk.DISKSTATUS_OK, 90*time.Second)
+
+	// Prepare image transfer request
+	transfersService := conn.SystemService().ImageTransfersService()
+	transfer := transfersService.Add()
+	image, err := ovirtsdk.NewImageBuilder().Id(diskId).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	imageTransfer, err := ovirtsdk.NewImageTransferBuilder().Image(
+		image,
+	).Direction(
+		ovirtsdk.IMAGETRANSFERDIRECTION_DOWNLOAD,
+	).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Initialize image transfer and lock the disk
+	transfer.ImageTransfer(imageTransfer)
+	transferReq, err := transfer.Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	currImageTransfer, ok := transferReq.ImageTransfer()
+
+	imageTransferId, ok := currImageTransfer.Id()
+
+	for {
+		if currImageTransfer.MustPhase() == ovirtsdk.IMAGETRANSFERPHASE_INITIALIZING {
+			time.Sleep(1 * time.Second)
+			currImageTransferReq, errIT := conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Get().Send()
+			if err != nil {
+				err = liberr.Wrap(errIT)
+				return
+			}
+			currImageTransfer, ok = currImageTransferReq.ImageTransfer()
+		} else {
+			break
+		}
+	}
+
+	caCert := secret.Data["cacert"]
+	caCertPool := x509.NewCertPool()
+	ok = caCertPool.AppendCertsFromPEM(caCert)
+	if !ok {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Create http client
+	tlsConfig := &tls.Config{
+		RootCAs: caCertPool,
+	}
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client := &http.Client{Transport: transport}
+
+	itURL, ok := currImageTransfer.ProxyUrl()
+
+	// creat regex expresions to filter the proxy URL
+	reInit := regexp.MustCompile(`https://`)
+	reUrl := regexp.MustCompile(`^[^/]+`)
+
+	transferURL := reInit.ReplaceAllString(itURL, "")
+	transferURL = reUrl.FindString(transferURL)
+	transferURL = fmt.Sprint("https://", transferURL, "/info/")
+
+	// Prepare disk GET request
+	req, err := http.NewRequest("GET", transferURL, nil)
+
+	// Run the request
+	resp, err := client.Do(req)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	defer resp.Body.Close()
+
+	//close image transfer
+	_, err = conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Cancel().Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		//retrive missing CA cert and update secrete
+		caUrl := fmt.Sprint(string(secret.Data["url"]), "ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA")
```

Comment:
suggestions regarding the ca retrieval:
I think it would be better initially to retrieve the CA and check the transfer with it, and in case it works compare it to the secret one and add if missing because in the current implementation we won't have any indication in the case of a failure with that automated retrieve CA.
Also should we handle or address somehow a failure (if a separated cert is confined for imagIO)?

Reply:
yeah, makes sense and when we think about the webhook we can to the following in the mutating webhook:
if there's connectivity to the source ovirt/rhv environment:
1. get the certificate
2. compare it against the one provided
3. if they are different, append the one that was retrieved to the specified one
and in the validation webhook:
if there's connectivity to the source ovirt/rhv environment:
1. test the ability to download the OVF store